src/components/BookList.js

```diff
@@ -1,11 +1,18 @@
 import './BookList.css';
-import { useSelector } from 'react-redux';
+import { useDispatch, useSelector } from 'react-redux';
+import { useEffect } from 'react';
 import Book from './Book';
+import { getBooks } from '../redux/books/booksSlice';
 
 const BookList = () => {
-  const { books } = useSelector((store) => store.books);
+  const { books, isLoading } = useSelector((store) => store.books);
+  const dispatch = useDispatch();
+  useEffect(() => {
+    dispatch(getBooks());
+  }, [books]);
```

Comment:
- [ ] You did a good job persisting the data on the API. :+1: There seems to be a small bug here, the app is fetching data infinitely because `books` is added as a dependency to the useEffect and resulting in an infinite loop. Kindly remove `books` from the dependency array to resolve the issue. 🧐️